### ISSUE TITLE
Missing CascadeAll on TreeChildren.d.ts

### ISSUE THREAD

**Issue by alessandro-p** [typeorm/typeorm#396]
First of all, congratulations for your work, it made my day. 

I was trying to build a hierarchical structure following the typeorm documentation, but when I tried to use the **CascadeAll** option on TreeChildren decorator as shown [here](https://typeorm.github.io/tables-and-columns.html#tree-tables) it gave me a type error since there is no CascadeAll option in types (See image below). 

![image](https://cloud.githubusercontent.com/assets/17408784/24902218/c4cd4a0e-1ea9-11e7-8aa4-7fd6fb35d46c.png)

So, since CascadeInsert and CascadeUpdate are provided, but CascadeAll is not specified, **I was wondering if the CascadeRemove behavior has not been implemented yet or if it is not expected at all.** 

Thanks in advance.

Alessandro

**Comment by pleerock** [comment]
There are lot of issues with cascades and especially cascade remove. Cascades are the most complex functionality of this framework. Right now I recommend you remove things manually without cascades until I fix all cascade-remove related problems

**Comment by imnotjames** [comment]
The docs at that page no longer exist but cascade on tree children has remove now.

**Comment by mateussilva92** [comment]
This issue might be fixed by #7156

**Comment by AlexMesser** [comment]
`TreeChildren` definition was changed to:
```ts
function TreeChildren(options?: { cascade?: boolean|("insert"|"update"|"remove"|"soft-remove"|"recover")[] })
```

here `cascade: true` is an equivalent of `cascadeAll`
